crypto/keyring/output.go

```diff
@@ -1,106 +1,73 @@
+// call to MkKeyOutput fails.
+func MkKeysOutput(infos []Info) ([]KeyOutput, error) {
 	kos := make([]KeyOutput, len(infos))
+	var err error
 	for i, info := range infos {
-		ko, err := Bech32KeyOutput(info)
+		kos[i], err = MkKeyOutput(info)
 		if err != nil {
 			return nil, err
 		}
-		kos[i] = ko
 	}
 
 	return kos, nil
 }
 
-// Bech32ConsKeyOutput create a KeyOutput in with "cons" Bech32 prefixes.
-func Bech32ConsKeyOutput(keyInfo Info) (KeyOutput, error) {
-	consAddr := sdk.ConsAddress(keyInfo.GetPubKey().Address().Bytes())
-
-	bechPubKey, err := sdk.Bech32ifyPubKey(sdk.Bech32PubKeyTypeConsPub, keyInfo.GetPubKey())
-	if err != nil {
-		return KeyOutput{}, err
-	}
-
-	return NewKeyOutput(keyInfo.GetName(), keyInfo.GetType().String(), consAddr.String(), bechPubKey), nil
+// MkConsKeyOutput create a KeyOutput in with "cons" Bech32 prefixes.
+func MkConsKeyOutput(keyInfo Info) (KeyOutput, error) {
+	pk := keyInfo.GetPubKey()
+	addr := sdk.ConsAddress(pk.Address().Bytes())
+	return NewKeyOutput(keyInfo.GetName(), keyInfo.GetType(), addr, pk)
 }
 
-// Bech32ValKeyOutput create a KeyOutput in with "val" Bech32 prefixes.
-func Bech32ValKeyOutput(keyInfo Info) (KeyOutput, error) {
-	valAddr := sdk.ValAddress(keyInfo.GetPubKey().Address().Bytes())
-
-	bechPubKey, err := sdk.Bech32ifyPubKey(sdk.Bech32PubKeyTypeValPub, keyInfo.GetPubKey())
-	if err != nil {
-		return KeyOutput{}, err
-	}
-
-	return NewKeyOutput(keyInfo.GetName(), keyInfo.GetType().String(), valAddr.String(), bechPubKey), nil
+// MkValKeyOutput create a KeyOutput in with "val" Bech32 prefixes.
+func MkValKeyOutput(keyInfo Info) (KeyOutput, error) {
+	pk := keyInfo.GetPubKey()
+	addr := sdk.ValAddress(pk.Address().Bytes())
+	return NewKeyOutput(keyInfo.GetName(), keyInfo.GetType(), addr, pk)
 }
 
-// Bech32KeyOutput create a KeyOutput in with "acc" Bech32 prefixes. If the
+// MkKeyOutput create a KeyOutput in with "acc" Bech32 prefixes. If the
 // public key is a multisig public key, then the threshold and constituent
 // public keys will be added.
-func Bech32KeyOutput(keyInfo Info) (KeyOutput, error) {
-	accAddr := sdk.AccAddress(keyInfo.GetPubKey().Address().Bytes())
-	bechPubKey, err := sdk.Bech32ifyPubKey(sdk.Bech32PubKeyTypeAccPub, keyInfo.GetPubKey())
-	if err != nil {
-		return KeyOutput{}, err
-	}
-
-	ko := NewKeyOutput(keyInfo.GetName(), keyInfo.GetType().String(), accAddr.String(), bechPubKey)
-
-	if mInfo, ok := keyInfo.(*multiInfo); ok {
-		pubKeys := make([]multisigPubKeyOutput, len(mInfo.PubKeys))
-
-		for i, pk := range mInfo.PubKeys {
-			accAddr := sdk.AccAddress(pk.PubKey.Address().Bytes())
-
-			bechPubKey, err := sdk.Bech32ifyPubKey(sdk.Bech32PubKeyTypeAccPub, pk.PubKey)
-			if err != nil {
-				return KeyOutput{}, err
-			}
-
-			pubKeys[i] = multisigPubKeyOutput{accAddr.String(), bechPubKey, pk.Weight}
-		}
-
-		ko.Threshold = mInfo.Threshold
-		ko.PubKeys = pubKeys
-	}
-
-	return ko, nil
+func MkKeyOutput(keyInfo Info) (KeyOutput, error) {
```

Comment:
```suggestion
func MakeKeyOutput(keyInfo Info) (KeyOutput, error) {
```

may be more readable

Reply:
I like `Mk` as well - it's meaning is pretty obvious and makes the code slightly shorter.

Reply:
What about 
```suggestion
func MkAccKeyOutput(keyInfo Info) (KeyOutput, error) {
```

? see https://github.com/cosmos/cosmos-sdk/pull/7477#discussion_r592298247

Reply:
Yeap. Good suggestion.